### ISSUE TITLE
Crash if server is stopped during startup

### ISSUE THREAD

**Issue by phako** [phako/dleyna-renderer#9]
```
^C=================================================================
==33654==ERROR: AddressSanitizer: heap-use-after-free on address 0x607000003620 at pc 0x7fe3ca108a81 bp 0x7ffe5ee43cd0 sp 0x7ffe5ee43cc8
READ of size 8 at 0x607000003620 thread T0
    #0 0x7fe3ca108a80 in prv_server_available_cb ../libdleyna/renderer/upnp.c:173
    #1 0x7fe3c97fa5cd in g_cclosure_marshal_VOID__OBJECTv ../../../../jhbuild/checkout/glib/gobject/gmarshal.c:1910
    #2 0x7fe3c97f7728 in _g_closure_invoke_va ../../../../jhbuild/checkout/glib/gobject/gclosure.c:873
    #3 0x7fe3c980f655 in g_signal_emit_valist ../../../../jhbuild/checkout/glib/gobject/gsignal.c:3406
    #4 0x7fe3c980f7f1 in g_signal_emit ../../../../jhbuild/checkout/glib/gobject/gsignal.c:3553
    #5 0x7fe3c97bfb9b in create_and_report_device_proxy ../../../../jhbuild/checkout/gupnp/libgupnp/gupnp-control-point.c:370
    #6 0x7fe3c97bfb9b in process_device_list ../../../../jhbuild/checkout/gupnp/libgupnp/gupnp-control-point.c:525
    #7 0x7fe3c97bfd59 in description_loaded ../../../../jhbuild/checkout/gupnp/libgupnp/gupnp-control-point.c:570
    #8 0x7fe3c97c0131 in got_description_url ../../../../jhbuild/checkout/gupnp/libgupnp/gupnp-control-point.c:629
    #9 0x7fe3c97c0131 in got_description_url ../../../../jhbuild/checkout/gupnp/libgupnp/gupnp-control-point.c:590
    #10 0x7fe3c96f52db in soup_session_process_queue_item ../../../../jhbuild/checkout/libsoup/libsoup/soup-session.c:2045
    #11 0x7fe3c96f5554 in soup_session_process_queue_item ../../../../jhbuild/checkout/libsoup/libsoup/soup-session.c:1972
    #12 0x7fe3c96f5554 in async_run_queue ../../../../jhbuild/checkout/libsoup/libsoup/soup-session.c:2085
    #13 0x7fe3c96f55d5 in idle_run_queue ../../../../jhbuild/checkout/libsoup/libsoup/soup-session.c:2112
    #14 0x7fe3ca002423 in g_main_dispatch ../../../../jhbuild/checkout/glib/glib/gmain.c:3344
    #15 0x7fe3ca002423 in g_main_context_dispatch ../../../../jhbuild/checkout/glib/glib/gmain.c:4062
    #16 0x7fe3ca002787 in g_main_context_iterate ../../../../jhbuild/checkout/glib/glib/gmain.c:4138
    #17 0x7fe3ca002a62 in g_main_loop_run ../../../../jhbuild/checkout/glib/glib/gmain.c:4336
    #18 0x7fe3c9fa419b in dleyna_main_loop_start ../libdleyna/core/main-loop.c:154
    #19 0x4011fb in main ../server/daemon.c:46
    #20 0x7fe3c9df5b74 in __libc_start_main (/lib64/libc.so.6+0x27b74)
    #21 0x4010dd in _start (/home/jens/Source/dleyna-renderer/build/server/dleyna-renderer-service+0x4010dd)

0x607000003620 is located 48 bytes inside of 72-byte region [0x6070000035f0,0x607000003638)
freed by thread T0 here:
    #0 0x7fe3ca1cca27 in free (/lib64/libasan.so.6+0xaea27)
    #1 0x7fe3ca109d2e in dlr_upnp_delete ../libdleyna/renderer/upnp.c:398
    #2 0x7fe3ca104b7e in prv_control_point_stop_service ../libdleyna/renderer/server.c:725
    #3 0x7fe3c9fa3da2 in prv_context_quit_cb ../libdleyna/core/main-loop.c:63
    #4 0x7fe3ca002423 in g_main_dispatch ../../../../jhbuild/checkout/glib/glib/gmain.c:3344
    #5 0x7fe3ca002423 in g_main_context_dispatch ../../../../jhbuild/checkout/glib/glib/gmain.c:4062

previously allocated by thread T0 here:
    #0 0x7fe3ca1cced7 in calloc (/lib64/libasan.so.6+0xaeed7)
    #1 0x7fe3ca008410 in g_malloc0 ../../../../jhbuild/checkout/glib/glib/gmem.c:136
    #2 0x7fe3ca1064ae in prv_control_point_start_service ../libdleyna/renderer/server.c:1131
    #3 0x7fe3c9fa3eb3 in prv_connector_acquired ../libdleyna/core/main-loop.c:97
    #4 0x7fe3c492252f in prv_bus_acquired ../src/connector-dbus.c:237
    #5 0x7fe3c994f2b1 in connection_get_cb ../../../../jhbuild/checkout/glib/gio/gdbusnameowning.c:504

SUMMARY: AddressSanitizer: heap-use-after-free ../libdleyna/renderer/upnp.c:173 in prv_server_available_cb
Shadow bytes around the buggy address:
  0x0c0e7fff8670: 00 00 00 00 fa fa fa fa 00 00 00 00 00 00 00 00
  0x0c0e7fff8680: 00 00 fa fa fa fa 00 00 00 00 00 00 00 00 00 00
  0x0c0e7fff8690: fa fa fa fa 00 00 00 00 00 00 00 00 00 00 fa fa
  0x0c0e7fff86a0: fa fa 00 00 00 00 00 00 00 00 00 00 fa fa fa fa
  0x0c0e7fff86b0: 00 00 00 00 00 00 00 00 00 00 fa fa fa fa fd fd
=>0x0c0e7fff86c0: fd fd fd fd[fd]fd fd fa fa fa fa fa 00 00 00 00
  0x0c0e7fff86d0: 00 00 00 00 00 00 fa fa fa fa 00 00 00 00 00 00
  0x0c0e7fff86e0: 00 00 00 00 fa fa fa fa fd fd fd fd fd fd fd fd
  0x0c0e7fff86f0: fd fa fa fa fa fa 00 00 00 00 00 00 00 00 00 00
  0x0c0e7fff8700: fa fa fa fa 00 00 00 00 00 00 00 00 00 00 fa fa
  0x0c0e7fff8710: fa fa 00 00 00 00 00 00 00 00 00 00 fa fa fa fa
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07 
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
  Shadow gap:              cc
==33654==ABORTING

[exited with 1]



```

**Comment by phako** [comment]
It looks like despite having all references to the CM dropped, the CP is still alive and has callbacks attached to it

**Comment by phako** [comment]
Caused by a "fix" in gupnp which added an additional reference to the CP when downloading the description document

**Comment by tapsiturtle** [comment]
Is it something you can fix on your side or has gupnp to be fixed?

**Comment by phako** [comment]
I am the GUPnP side ;) . Just waiting for some feedback over there.

**Comment by phako** [comment]
Fixed in GUPnP 1.2.7
